src/LinearSolverTypes.f90

```diff
@@ -755,6 +775,54 @@ SUBROUTINE updatedA(solver)
   solver%isDecomposed=.FALSE.
 
 ENDSUBROUTINE updatedA
+
+SUBROUTINE getResidualVec_LinearSolverType_Direct(solver,resid,nIters)
+  CHARACTER(LEN=*),PARAMETER :: myName='getResidualVec_LinearSolverType_Direct'
+  CLASS(LinearSolverType_Direct),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: resid
+  INTEGER(SIK),INTENT(OUT),OPTIONAL :: nIters
+
+  REQUIRE(solver%isInit)
+  REQUIRE(resid%isInit)
+  REQUIRE(ASSOCIATED(solver%b))
+  REQUIRE(solver%b%isInit)
+  REQUIRE(ASSOCIATED(solver%A))
+  REQUIRE(solver%A%isInit)
+  REQUIRE(ASSOCIATED(solver%x))
+  REQUIRE(solver%x%isInit)
+
+  IF (resid%isInit .AND. solver%isInit) THEN
+    CALL BLAS_copy(solver%b,resid)
+    CALL BLAS_matvec(thisMatrix=solver%A,alpha=-1.0_SRK,x=solver%x,beta=1.0_SRK,y=resid)
+  ELSE
+    CALL eLinearSolverType%raiseError('Incorrect call to '// &
+       modName//'::'//myName//' - Residual or solver not initialized.')
+  ENDIF
+
+  ! Direct method; nIters has no meaning
+  IF (PRESENT(nIters)) nIters = 0
```

Comment:
Should direct methods return 0 iterations or a known negative value to indicate the method was direct?

Reply:
It's an open question; I figured 0 would reflect the direct nature, as I figured a negative value is typically used as a flag that an error occurred. @djabaay do you have any thoughts?

Reply:
My thought is, the default value is zero, so if an error happens that doesn't change the default value (like the method itself is somehow accidentally skipped), then you don't necessarily have an indication of an error. But if direct methods always returned like a -151 (or whatever) when they finished, then you can be more certain an error didn't occur in the solve.

Reply:
Personally I think 0 is fine here